core/shared/src/main/scala/org/scalacheck/Prop.scala

```diff
@@ -788,7 +788,7 @@ object Prop {
     def shrinker(x: T, r: Result, shrinks: Int, orig: T): Result = {
       val xs = shrink(x)
       val res = r.addArg(Arg(labels,x,shrinks,orig,pp(x),pp(orig)))
-      val originalException = Some(r.status).collect { case NonFatal(e) => e.getClass() }
+      val originalException = Some(r.status).collect { case Exception(e) => e.getClass }
```

Comment:
I stared at this yesterday and today and pulled up the PR on my local and added `println`s before I finally understood why such an innocent looking change makes a huge difference 😂

This is very confusing. The `Exception` here is _NOT_ in the `Throwable` hierarchy. That's why `NonFatal(_)` was effectively useless.
https://github.com/typelevel/scalacheck/blob/4867326242450a54f66a5ecbccc2a59dc4630a64/core/shared/src/main/scala/org/scalacheck/Prop.scala#L300

Oddly enough I don't think scalac warns for this?

Reply:
Perhaps if you write it as `Prop.Exception` that will help with readability? Perhaps the reviewer in https://github.com/typelevel/scalacheck/pull/838#discussion_r913161658 was confused as well 😝 

Also I now realize you explained exactly this in https://github.com/typelevel/scalacheck/pull/838#discussion_r931864166 but I obviously didn't understand 😂 sorry about that!

Reply:
I think we were all confused by the name 😝. Good point about using `Prop.Exception`. It will make clear we're not talking about the usual `Exception`. I'll update it (it's not the only match on `Prop.Exception` in the file though).

I think scalac infer `{ case NonFatal(e) => e.getClass() }` as a `PartialFunction[Any, Class[_ <: Throwable]]` since that's what makes it compile, so it does not give a warning :-( . Intellij points to the error though. Unfortunately I was not using it in my first PR.